### DISCUSSION TITLE
Vercel template doesn't support defer/streaming

### DISCUSSION THREAD

**Discussion by luisrudge** [remix-run/remix#5359] [Q&A] [answered]
It currently uses `renderToString` which doesn't support streaming.

https://github.com/remix-run/remix/blob/b928040061890a6ef0270abdb5b1201638f0dd00/templates/vercel/app/entry.server.tsx

**Comment by sergiodxa** [comment, marked as answer]
The Vercel adapter doesn't use their Edge platform and because of this it can't support SSR Streaming since Vercel when not using Edge uses AWS Lambda which doesn't support it.

The code could be updated to do SSR streaming, but the platform will buffer the response until it's ready anyway.

When Vercel Edge platform is supported as an adapter it should be possible to do SSR Streaming.

**Comment by luisrudge** [comment]
@sergiodxa oh! I had no idea. Thanks for clarifying. I didn't see this mentioned in the docs.

**Comment by nicksrandall** [comment]
Vercel does have some magic in their node runtime to make streaming work (this is how Next.js has support for streaming when using Node). What is unclear to me is if somebody from Vercel has to open this up for Remix or is somebody on Remix team needs to build something to tap into that behavior.

On a related note, Jacob Ebby has a POC of using Lambda + Cloudflare + Websockets (very similar to what Vercel is doing) to get show that streaming rendering is possible with Remix in such an environment.

More context: https://twitter.com/leeerob/status/1618704234899451904

**Comment by WesleyYue** [comment]
They're working on it! https://github.com/vercel/vercel/pull/9375
